src/internal_events/redis.rs

```diff
@@ -0,0 +1,91 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct RedisEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for RedisEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Received one event.", rate_limit_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_events_total", 1);
+        counter!("processed_bytes_total", self.byte_size as u64);
```

Comment:
```suggestion
        counter!("processed_events_total", 1);
        counter!("processed_bytes_total", self.byte_size as u64);
```

We should drop these and only count them in `RedisEventSent`. Otherwise these events will be double counted.

We have logic at the topology level that will add an `events_in_total` that will indicate the number of events that have gone into the sink.